src/scss/_type.scss

```diff
@@ -109,131 +109,213 @@
   }
 }
 
-@mixin text($token) {
-  // Labels
-  @if $token == 'button' {
-    @include labelText($button);
-  }
+@mixin modifying-styles($font-size, $font-weight: 400, $text-transform: unset, $font-style: unset) {
+  font-size: $font-size;
+  font-style: $font-style;
+  font-weight: $font-weight;
+  letter-spacing: 0;
+  text-transform: $text-transform;
+}
 
-  @if $token == 'link' {
-    @include labelText($link);
-  }
+@mixin display-styles($font-size, $font-weight: 400, $text-transform: unset, $font-style: unset) {
+  @include DistinctDisplay;
 
-  @if $token == 'email' {
-    @include labelText($email);
-  }
+  line-height: calc($font-size + $line-height-m-units);
 
-  @if $token == 'label' {
-    @include labelText($label);
-  }
+  @include modifying-styles($font-size, $font-weight, $text-transform, $font-style);
+}
 
-  @if $token == 'badge' {
-    @include labelText($badge);
-  }
+@mixin heading-styles($font-size, $font-weight: 400, $text-transform: unset, $font-style: unset) {
+  @include display-styles($font-size, $font-weight, $text-transform, $font-style);
+
+  line-height: calc($font-size + $line-height-m-units);
+}
+
+@mixin body-styles($font-size, $font-weight: 400, $text-transform: unset, $font-style: unset) {
+  line-height: calc($font-size + $line-height-l-units);
+
+  @include modifying-styles($font-size, $font-weight, $text-transform, $font-style);
+  @include Montserrat;
+}
+
+@mixin link-styles(
+  $font-size,
+  $font-weight: 600,
```

Comment:
we only want to use `font-weight` for non dynamic fonts, we need to use `font-variation-settings` for dynamic fonts